tests/integration/devfile/cmd_devfile_debug_test.go

```diff
@@ -147,16 +113,36 @@ var _ = Describe("odo devfile debug command tests", func() {
 			if runtime.GOOS != "windows" {
 				Expect(helper.ListFilesInDir(os.TempDir())).To(Not(ContainElement(commonVar.Project + "-app" + "-nodejs-cmp-" + commonVar.Project + "-odo-debug.json")))
 			}
+		})
+		It("should start a debug session and run debug info on a running debug session", func() {
+			httpPort, err := util.HTTPGetFreePort()
+			Expect(err).NotTo(HaveOccurred())
+			freePort := strconv.Itoa(httpPort)
 
+			stopChannel := make(chan bool)
+			go func() {
+				helper.Cmd("odo", "debug", "port-forward", "--local-port", freePort, "--context", commonVar.Context).WithTerminate(60*time.Second, stopChannel).ShouldRun()
+			}()
+
+			// 400 response expected because the endpoint expects a websocket request and we are doing a HTTP GET
+			// We are just using this to validate if nodejs agent is listening on the other side
+			helper.HttpWaitForWithStatus("http://localhost:"+freePort, "WebSockets request was expected", 12, 5, 400)
+			runningString := helper.Cmd("odo", "debug", "info", "--context", commonVar.Context).ShouldPass().Out()
+			Expect(runningString).To(ContainSubstring(freePort))
+			Expect(helper.ListFilesInDir(os.TempDir())).To(ContainElement(commonVar.Project + "-app-" + componentName + "-odo-debug.json"))
+			defer helper.DeleteFile(filepath.Join(os.TempDir(), commonVar.Project+"-app-"+componentName+"-odo-debug.json"))
+			stopChannel <- true
 		})
+
 	})
 
-	Context("when the debug command throws an error during push", func() {
-		It("should wait and error out with some log", func() {
+	When("the debug command throws an error during push", func() {
+		BeforeEach(func() {
 			helper.Cmd("odo", "create", "--project", commonVar.Project, "--context", commonVar.Context, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-debugrun.yaml")).ShouldPass()
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
 			helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "npm run debug", "npm run debugs")
```

Comment:
Can you move this to the first `When` block? So the structure would look something like this:
```go
When("creating an application using devfile with debugrun", func() {
	BeforeEach(func() {
		helper.Cmd("odo", "create", "--project", commonVar.Project, componentName, "--context", commonVar.Context, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-debugrun.yaml")).ShouldPass()
		helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
	})
	When("the debug command in the devfile is invalid", func() {
		BeforeEach(func() {
			helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "npm run debug", "npm run debugs")
		})
		It("should return an error message along with a log on odo push", func() {})
	})
	When("the application is pushed", func() {
		BeforeEach(func() {
			helper.Cmd("odo", "push", "--debug", "--context", commonVar.Context).ShouldPass()
		})
		It("checks that machine output debug information works", func() {})

		It("should expect a ws connection when tried to connect on default debug port locally", func() {})

		It("should error out when using an invalid devfile", func() {})

		It("should start a debug session and run debug info on a closed debug session", func() {})
		It("should start a debug session and run debug info on a running debug session", func() {})
	})
})
```

Let me know if you think otherwise :)